pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -108,43 +125,74 @@ class CollectionViewSet(
     serializer_class = CollectionSerializer
     pagination_class = LimitOffsetPagination
 
+    def filter_queryset(self, queryset):
+        """
+        Filter Repository related fields.
+        """
+        queryset = super().filter_queryset(queryset)
+
+        try:
+            user_deprecated_value = self.request.query_params["deprecated"]
+        except MultiValueDictKeyError:
+            pass
+        else:
+            if user_deprecated_value.lower() in ["true", "yes", "1"]:
+                queryset = queryset.filter(deprecated=True)
+            elif user_deprecated_value.lower() in ["false", "no", "0"]:
+                queryset = queryset.filter(deprecated=False)
+            else:
+                raise ValidationError("Cannot parse value of `deprecated` GET parameter")
+
+        return queryset
+
     def get_queryset(self):
         """
-        Returns a CollectionVersions queryset for specified distribution.
+        Returns a Collections queryset for specified distribution.
```

Comment:
In looking closer, `galaxy_ng` relies on this viewset returning a queryset for `CollectionVersion` not `Collection`. We support searching across `CollectionVersion`s in the collection list endpoint - previously via `galaxy_pulp` bindings, more recently by using the [CollectionVersionFilter](https://github.com/pulp/pulp_ansible/blob/master/pulp_ansible/app/viewsets.py#L106) in the `galaxy_ng` `CollectionViewSet` used by the ui.

If we have this return a queryset for `Collection`, any thoughts on how to support search?

Reply:
What are some examples of these API calls? Also do you really mean search or do you mean filter?

Reply:
We filter, based on the user’s search query. Examples:
```
/api/automation-hub/_ui/v1/repo/published/?keywords=nginx
/api/automation-hub/_ui/v1/repo/published/?q=nginx
/api/automation-hub/_ui/v1/repo/rh-certified/?tags=aws
```

Reply:
Using the annotation setup, this seems to be working in `galaxy_ng`

Reply:
That's great!